plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -925,7 +964,8 @@ private static ObjectReadFunction longTimestampWithTimeZoneReadFunction()
         return ObjectReadFunction.of(
                 LongTimestampWithTimeZone.class,
                 (resultSet, columnIndex) -> {
-                    OffsetDateTime offsetDateTime = resultSet.getObject(columnIndex, OffsetDateTime.class);
+                    String stringValue = resultSet.getString(columnIndex);
+                    OffsetDateTime offsetDateTime = ZonedDateTime.from(DATE_TIME_OFFSET_FORMATTER.parse(stringValue)).toOffsetDateTime();
```

Comment:
is this change and the one above in `shortTimestampWithTimeZoneReadFunction` strictly required in this PR?
I'm unable to understand the significance probably.

Reply:
Yes, the issue is the driver changes from their internal `DateTimeOffset` class to the `java.time.OffsetDateTime`, the date shifts from what is stored. This instead gets the string representation of the date and we parse it to `OffsetDateTime`.

https://github.com/microsoft/mssql-jdbc/blob/main/src/main/java/microsoft/sql/DateTimeOffset.java#L229-L234

See https://github.com/microsoft/mssql-jdbc/issues/2246 for examples and their discussion. The root cause of this seems to be due to skipping days/dates when the Gregorian calendar switched. The MSSQL issue is closed, likely nothing that is going to be fixed because maybe it isn't broken? Dates are fun.